services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaController.kt

```diff
@@ -71,4 +80,124 @@ class HankeKayttajaController(
 
         return HankeKayttajaResponse(users)
     }
+
+    @PutMapping
+    @Operation(
+        summary = "Update permissions of the listed users.",
+        description =
+            """
+For each user, permissions are updated directly to the active permissions
+if the user in question has activated his permissions. If they haven't,
+permissions are updated to the user tokens, i.e. this call will change
+which permissions the user will get when they activate their permissions.
+
+The reply from the list users endpoint can be modified and used here, but
+only the ID and role fields are read.
+
+Only the IDs that are mentioned are updated, so only updated IDs need to
+be listed. This also means that this endpoint can't be used to remove
+permissions.
+
+If removing or adding KAIKKI_OIKEUDET role, the caller needs to have those
+same permissions.
+"""
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(description = "Success, no response body", responseCode = "204"),
+                ApiResponse(
+                    description = "The request body was invalid",
+                    responseCode = "400",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+                ApiResponse(
+                    description = "Not authorized to change admin permissions",
+                    responseCode = "403",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+                ApiResponse(
+                    description = "Hanke by requested tunnus not found",
+                    responseCode = "404",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+                ApiResponse(
+                    description =
+                        "User tried editing their own permissions or there would " +
+                            "be no users with KAIKKI_OIKEUDET left.",
+                    responseCode = "409",
+                    content = [Content(schema = Schema(implementation = HankeError::class))]
+                ),
+            ]
+    )
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    fun updatePermissions(
+        @ValidHanke @RequestBody permissions: PermissionUpdate,
+        @PathVariable hankeTunnus: String
+    ) {
+        // featureFlags.ensureEnabled(Feature.PERMISSION_EDITING)
```

Comment:
Unused code should be deleted and can be retrieved from source control history if required.

Reply:
Oh yeah, I meant to ask about this, whether we need a feature flag for this. I'm guessing we still have some time until we're ready to release the user management feature.

Reply:
Hmm, I guess we should follow the same procedure as with hanke. So let's use a feature flag for edit endpoints.

Reply:
I added the feature flag.

In another commit, I went a bit further and used the feature flag to stop the creation of user tokens until we are ready with the user management feature. The tokens really have any meaning if we don't send the invitation emails.